### DISCUSSION TITLE
Installing the SDK on a Raspberry Pi

### DISCUSSION THREAD

**Discussion by bartbutenaers** [phoddie/node-red-mcu#64] [Q&A] [answered]
Hi everybody,

I have followed the Linux instructions to install all the required tools on my Raspberry Pi 4.  Note that it runs Raspberry Pi OS ***64 bit***, which has not been tested yet according to your [documentation](https://github.com/Moddable-OpenSource/moddable/blob/public/documentation/Moddable%20SDK%20-%20Getting%20Started.md#linux).  I need the 64 bit to run experiments on my Google Coral Tpu stick, so unfortunately I cannot downgrade to 32 bit. 

I had 2 issues during the installation process:
1.  The `sudo apt-get install gcc git wget make libncurses-dev flex bison gperf` did give me an error.  So I decided to run `sudo apt-get update`, and the second time the error was gone.  So it might be useful to add this command to the documentation.

2. I had no errors during the next step, until I executed this one:
   ```
   pi@raspberrypi:~/Projects/moddable/build/makefiles/lin $ xsbug
   Unable to init server: Could not connect: Connection refused

   (xsbug:12681): Gtk-WARNING **: 23:22:13.014: cannot open display:
   ```
   Do you have any tips of things I could try?  Or do I need to install something extra before executing this command perhaps?

Thanks (again)!
Bart

**Comment by phoddie** [comment]
@bartbutenaers – Let's see if we can figure this out. We definitely do run on some 64-bit Linux targets, so it is likely possible on Raspberry Pi too.

Linux is outside my expertise, so I defer to @mkellner and @acarle on the details.

FWIW – Strictly speaking this isn't an issue with Node-RED MCU Edition but the underlying Moddable SDK. That can sometimes be confusing to sort out. For future reference, it would be best to post questions like this on the [Moddable SDK repository](http://github.com/moddable-OpenSource/moddable).

**Comment by mkellner** [comment, marked as answer]
@bartbutenaers, xsbug is a Gtk application and requires a windowing system to run. If you do have a windows system set up on your Pi 4, you can try setting the DISPLAY environment variable with: `export DISPLAY=:0.0` and try running xsbug again.

**Comment by bartbutenaers** [comment]
@phoddie: Ah yes, of course I should have posted this issue in the other repo.  Sorry for the confusion!

@mkellner: That makes everything clear.  I am using my Raspberry headless, and I access it only via Node-RED (and Putty).
Perhaps it would be useful to add your note about this in the documentation (where the xsbug command is called), to avoid that beginners like me ask themselves the same silly question.  Because I think a lot of Node-RED users use a similar setup as mine: some time ago there was a [survey](https://nodered.org/about/community/survey/2019/), and it seemed that most users in the Node-RED community use Raspberry.  I assume a lot of these installations will be headless, because you can administrate it mostly via the Node-RED web interface. 

Thanks!!  Case closed.

**Reply by phoddie** [reply]
@bartbutenaers – no problem at all. 

FWIW – the Moddable team has been exploring how we can better support headless development machines. It hasn't been a priority, but I can understand why Node-RED users operate that way. We hope to have a first step you can try out later this week. No doubt it will be a process of refinement based on experience to arrive at a complete solution. Your continued feedback will certainly help that along.
